Source/PangleAdapter.swift

```diff
@@ -74,37 +78,54 @@ final class PangleAdapter: PartnerAdapter {
         completion(.success([:]))
     }
     
-    /// Indicates if GDPR applies or not and the user's GDPR consent status.
-    /// - parameter applies: `true` if GDPR applies, `false` if not, `nil` if the publisher has not provided this information.
-    /// - parameter status: One of the `GDPRConsentStatus` values depending on the user's preference.
-    func setGDPR(applies: Bool?, status: GDPRConsentStatus) {
+    /// Indicates that the user consent has changed.
+    /// - parameter consents: The new consents value, including both modified and unmodified consents.
+    /// - parameter modifiedKeys: A set containing all the keys that changed.
+    func setConsents(_ consents: [ConsentKey: ConsentValue], modifiedKeys: Set<ConsentKey>) {
         // See PAGConfig.gdprConsent documentation on PAGConfig.h
-        if applies == true {
-            let gpdrConsent: PAGGDPRConsentType = status == .granted ? .consent : .noConsent
-            PAGConfig.share().gdprConsent = gpdrConsent
-            log(.privacyUpdated(setting: "gdprConsent", value: gpdrConsent.rawValue))
+        if modifiedKeys.contains(partnerID) || modifiedKeys.contains(ConsentKeys.gdprConsentGiven) {
+            let consent = consents[partnerID] ?? consents[ConsentKeys.gdprConsentGiven]
+            switch consent {
+            case ConsentValues.granted:
+                let gdprConsent = PAGGDPRConsentType.consent
+                PAGConfig.share().gdprConsent = gdprConsent
+                log(.privacyUpdated(setting: "gdprConsent", value: gdprConsent.rawValue))
+            case ConsentValues.denied:
+                let gdprConsent = PAGGDPRConsentType.noConsent
+                PAGConfig.share().gdprConsent = gdprConsent
+                log(.privacyUpdated(setting: "gdprConsent", value: gdprConsent.rawValue))
+            default:
+                break   // do nothing
+            }
+        }
+
+        // See PAGConfig.doNotSell documentation on PAGConfig.h
+        if modifiedKeys.contains(ConsentKeys.ccpaOptIn) {
+            let consent = consents[ConsentKeys.ccpaOptIn]
+            switch consent {
+            case ConsentValues.granted:
+                let doNotSell = PAGDoNotSellType.sell
+                PAGConfig.share().doNotSell = doNotSell
+                log(.privacyUpdated(setting: "doNotSell", value: doNotSell.rawValue))
```

Comment:
```suggestion
                PAGConfig.share().doNotSell = .sell
                log(.privacyUpdated(setting: "doNotSell", value: PAGDoNotSellType.sell.rawValue))
```

It feels slightly more clear to me if we just have a slight bit of duplication here, as the constant `doNotSell` being set to `sell` is kind of odd to me. Same goes as above for GDPR.

We could also name the constant something like `doNotSellValue`, which might make it more clear.

Reply:
yeah makes sense. I simplified everything a bit without variables